dashboard/src/components/DeploymentFormBody/DeploymentFormBody.tsx

```diff
@@ -148,13 +149,15 @@ class DeploymentFormBody extends React.Component<
     // TODO(andres): This requires refactoring. Currently, the deploy and upgrade
     // forms behave differently. In the deployment form, a change in the version
     // changes the route but in the case of the upgrade it only changes the state
-    const { namespace, selected, deploymentEvent } = this.props;
+    const { cluster, namespace, selected, deploymentEvent } = this.props;
 
     if (deploymentEvent === "upgrade") {
       const { chartID, chartNamespace, getChartVersion } = this.props;
       getChartVersion(chartNamespace, chartID, e.currentTarget.value);
     } else {
-      this.props.push(url.app.apps.new(selected.version!, namespace, e.currentTarget.value));
+      this.props.push(
```

Comment:
`getChartVersion` above will need the `cluster` param as well, isn't it?

Reply:
I'm doing the changes systematically in consumable PRs. First going through and making sure all the UI urls correctly use the cluster (rather than just using the "default"), then I'll do the backend calls.